src/test/java/uk/gov/justice/hmpps/prison/util/WithMockAuthUser.kt

```diff
@@ -0,0 +1,66 @@
+package uk.gov.justice.hmpps.prison.util
+
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.test.context.support.WithSecurityContext
+import org.springframework.security.test.context.support.WithSecurityContextFactory
+import org.springframework.util.Assert
+import uk.gov.justice.hmpps.kotlin.auth.AuthAwareAuthenticationToken
+import uk.gov.justice.hmpps.kotlin.auth.AuthSource
+import java.lang.annotation.Inherited
+
+@Target(
+  AnnotationTarget.FUNCTION,
+  AnnotationTarget.PROPERTY_GETTER,
+  AnnotationTarget.PROPERTY_SETTER,
+  AnnotationTarget.CLASS,
+)
+@Retention(AnnotationRetention.RUNTIME)
+@Inherited
+@WithSecurityContext(factory = WithMockUserSecurityContextFactory::class)
+annotation class WithMockAuthUser(
+  val value: String = "user",
+  val username: String = "",
+  val authorities: Array<String> = [],
+  val roles: Array<String> = ["USER"],
+  val authSource: AuthSource = AuthSource.NONE,
+  val clientId: String = "prison-api-user",
+)
+
+internal class WithMockUserSecurityContextFactory() : WithSecurityContextFactory<WithMockAuthUser> {
```

Comment:
copied from Spring WithMockUserSecurityContextFactory but altered to put `AuthAwareAuthenticationToken` in context instead